pxr/usd/sdf/copyUtils.cpp

```diff
@@ -61,30 +61,50 @@ namespace {
 
 } // end anonymous namespace
 
-// Returns lists of value and children field names to be handled during
+// Returns lists of value field names to be handled during
 // the copy process. The returned lists are sorted using the
 // TfTokenFastArbitraryLessThan comparator.
 static
-void
-_GetFieldNames(
-    const SdfLayerHandle& layer, const SdfPath& path, 
-    std::vector<TfToken>* valueFields, 
-    std::vector<TfToken>* childrenFields)
+std::vector<TfToken>
+_GetValueFieldNames(
+    const SdfLayerHandle& layer, const SdfPath& path)
 {
+    std::vector<TfToken> valueFields;
     const SdfSchemaBase& schema = layer->GetSchema();
     const std::vector<TfToken> allFields = layer->ListFields(path);
     for (const TfToken& field : allFields) {
-        if (schema.HoldsChildren(field)) {
-            childrenFields->push_back(field);
+        if (!schema.HoldsChildren(field)) {
+            valueFields.push_back(field);
         }
-        else {
-            valueFields->push_back(field);
+    }
+
+    TfTokenFastArbitraryLessThan lessThan;
+    std::sort(valueFields.begin(), valueFields.end(), lessThan);
+
+    return valueFields;
+}
+
+// Returns lists of children field names to be handled during
+// the copy process. The returned lists are sorted using the
+// TfTokenFastArbitraryLessThan comparator.
+static
+std::vector<TfToken>
+_GetChildrenFieldNames(
+    const SdfLayerHandle& layer, const SdfPath& path)
+{
+    std::vector<TfToken> childrenFields;
+    const SdfSchemaBase& schema = layer->GetSchema();
+    const std::vector<TfToken> allFields = layer->ListFields(path);
+    for (const TfToken& field : allFields) {
```

Comment:
Minor suggestion.  If you use `std::remove_if` / `erase`, it'll avoid the extra copies of the tokens.

```
std::vector<TfToken> fields = layer->ListFields(path);
// In C++20, this can be just erase_if
fields.erase(std::remove_if(std::begin(fields), std::end(fields), [&schema](const auto& field) { return schema.HoldsChildren(field); }, std::end(fields));
```